src/lists.rs

```diff
@@ -41,11 +47,38 @@ pub struct ListFormatting<'a> {
     pub h_width: usize,
     // Available width if we layout vertically
     pub v_width: usize,
+    // Non-expressions, e.g. items, will have a new line at the end of the list.
+    // Important for comment styles.
+    pub is_expression: bool
```

Comment:
The more I think about this, the less I like it. Could we append a newline and block_indent instead of using a block comment in the expression case? It feels a bit irregular atm.